depthai_nodes/parsing_neural_network.py

```diff
@@ -127,6 +127,13 @@ def setBackend(self, setBackend: str) -> None:
         """Sets the backend of the NeuralNetwork node."""
         self._nn.setBackend(setBackend)
 
+    def getParser(self, parserID: int = 0) -> BaseParser:
+        """Returns the parser node for the specified model head."""
+        assert (
```

Comment:
Let's go with if check and KeyError because assert can be tehnically ommited if you run the code in optimized mode (-O flag)